pkg/controller/controlplane/actuator.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved.
+// This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener-extension-provider-equinix-metal/pkg/imagevector"
+	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
+	"github.com/gardener/gardener/extensions/pkg/controller/controlplane"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/utils/chart"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+)
+
+const (
+	namespacesChartName = "namespaces"
+)
+
+// NewActuator creates a new Actuator that acts upon and updates the status of ControlPlane resources.
+func NewActuator(
+	a controlplane.Actuator,
+	additionalCharts []*chart.Chart,
+	chartRendererFactory extensionscontroller.ChartRendererFactory,
+) controlplane.Actuator {
+	return &actuator{
+		Actuator:             a,
+		additionalCharts:     additionalCharts,
+		chartRendererFactory: chartRendererFactory,
+	}
+}
+
+// actuator is a controlplane.Actuator that acts upon and updates the status of ControlPlane resources.
+// This one runs the normal control plane activities, and then also applies the additional charts, if any
+// chart.
+type actuator struct {
+	controlplane.Actuator
+	client                         client.Client
+	chartRendererFactory           extensionscontroller.ChartRendererFactory
+	additionalChartsValuesProvider valuesProvider
```

Comment:
I don't see this field initialized anywhere, I guess it should be initialized via the constructor. Also I think a shorter name such as `valuesProvider` is more appropriate.